### ISSUE TITLE
SkeletonView not working in TableViewCell with custom class

### ISSUE THREAD

**Issue by pennywise94** [Juanpe/SkeletonView#125]
### What did you do?
Changed the UITableViewDataSource to SkeletonTableViewDataSource
Returned the correct Cell Identifier
Changed estimatedRowHeight to a fixed height (150), as I am using automatic row height
Made all my views skeletonable, including the UITableView and the cell themselves
Tried to set myTableView.showSkeleton() on viewDidAppear() and in a custom button, both with the same unexpected result

### What did you expect to happen?
I am successfully using SkeletonView while loading cells of a UICollectionView, so I expected the cells to show skeletons as well.

### What happened instead?
I only see one grey solid where the UITableView should appear. Not the views of my cell, only a grey solid.

![Unexpected result](https://i.imgur.com/pkk81b2.png)

I then researched and found another, similar issue of someone who was unable to get the subviews in the debugger, which happens for me as well. Nothing underneath the FollowCell seems to be skeletonable, while everything is set up correctly:

![Debug](https://i.imgur.com/6QfiSMw.png)

There are no cells in my view hierarchy that are blocking the skeletons (everything has isSkeletonable = true).

### Steps to reproduce the behavior
import SkeletonView
add UITableView as a subview to a UIViewController
register cell with custom class
make all views skeletonable
use showSkeleton() to show the skeletons (in viewDidAppear)

### SkeletonView Environment
**SkeletonView version:** 1.4.1
**Xcode version:** 10.1
**Swift version:** 4.2

**Comment by pennywise94** [comment]
Any tips, ideas or fixes for this? I have come across this issue again, but would love to use SkeletonView for my UITableViews. Figured out that it probably has something to do with the contentView of the UITableViewCells - could that be? As you can see in the debug screenshot above, there are no subSkeletons shown under the UITableViewCells, which makes me think something (the contentView?) is blocking it. However, in the initializer of the custom UITableView class, I set contentView.isSkeletonable to true.

**Comment by jpeckner** [comment]
@pennywise94 I ran into this issue too; the giant solid happens because calling `showSkeleton()` on a `UITableView` calls `addDummyDataSource()`, which calls `reloadData()`, which immediately afterwards results in `visibleCells` returning an empty array. However, it turns out that [only visibleCells are skeleton'd](https://github.com/Juanpe/SkeletonView/blob/1671af6d2805211dc258f4b9b8574d228dd9efd7/Sources/SubviewsSkeletonables.swift#L18), so that explains why nothing is recursively skeleton'd below the `UITableView`. This definitely seems like a bug; if I get a chance I'll submit a PR to fix it.

In the meantime, instead of calling `tableView.showSkeleton()` in `viewDidAppear()`, `tableView.visibleCells.forEach { $0.showSkeleton() }` worked for me. Let me know how that goes!

**Comment by pennywise94** [comment]
> @pennywise94 I ran into this issue too; the giant solid happens because calling `showSkeleton()` on a `UITableView` calls `addDummyDataSource()`, which calls `reloadData()`, which immediately afterwards results in `visibleCells` returning an empty array. However, it turns out that [only visibleCells are skeleton'd](https://github.com/Juanpe/SkeletonView/blob/1671af6d2805211dc258f4b9b8574d228dd9efd7/Sources/SubviewsSkeletonables.swift#L18), so that explains why nothing is recursively skeleton'd below the `UITableView`. This definitely seems like a bug; if I get a chance I'll submit a PR to fix it.
> 
> In the meantime, instead of calling `tableView.showSkeleton()` in `viewDidAppear()`, `tableView.visibleCells.forEach { $0.showSkeleton() }` worked for me. Let me know how that goes!

Hi mate, thanks for the reply. I eventually figured out that it had to do with the contentView for me. I added all my constraints to the contentView rather than the cell itself and it works like that. I also create a function addSkeleton() which is called in viewDidLayoutSubviews() and this works just fine for me.

**Comment by erfanwakka** [comment]
> > @pennywise94 I ran into this issue too; the giant solid happens because calling `showSkeleton()` on a `UITableView` calls `addDummyDataSource()`, which calls `reloadData()`, which immediately afterwards results in `visibleCells` returning an empty array. However, it turns out that [only visibleCells are skeleton'd](https://github.com/Juanpe/SkeletonView/blob/1671af6d2805211dc258f4b9b8574d228dd9efd7/Sources/SubviewsSkeletonables.swift#L18), so that explains why nothing is recursively skeleton'd below the `UITableView`. This definitely seems like a bug; if I get a chance I'll submit a PR to fix it.
> > In the meantime, instead of calling `tableView.showSkeleton()` in `viewDidAppear()`, `tableView.visibleCells.forEach { $0.showSkeleton() }` worked for me. Let me know how that goes!
> 
> Hi mate, thanks for the reply. I eventually figured out that it had to do with the contentView for me. I added all my constraints to the contentView rather than the cell itself and it works like that. I also create a function addSkeleton() which is called in viewDidLayoutSubviews() and this works just fine for me.

Could you explain more what did you do ?

**Comment by adelbios** [comment]
> > @pennywise94 I ran into this issue too; the giant solid happens because calling `showSkeleton()` on a `UITableView` calls `addDummyDataSource()`, which calls `reloadData()`, which immediately afterwards results in `visibleCells` returning an empty array. However, it turns out that [only visibleCells are skeleton'd](https://github.com/Juanpe/SkeletonView/blob/1671af6d2805211dc258f4b9b8574d228dd9efd7/Sources/SubviewsSkeletonables.swift#L18), so that explains why nothing is recursively skeleton'd below the `UITableView`. This definitely seems like a bug; if I get a chance I'll submit a PR to fix it.
> > In the meantime, instead of calling `tableView.showSkeleton()` in `viewDidAppear()`, `tableView.visibleCells.forEach { $0.showSkeleton() }` worked for me. Let me know how that goes!
> 
> Hi mate, thanks for the reply. I eventually figured out that it had to do with the contentView for me. I added all my constraints to the contentView rather than the cell itself and it works like that. I also create a function addSkeleton() which is called in viewDidLayoutSubviews() and this works just fine for me.

thank u... finally it's work 😍😍😍

**Comment by pennywise94** [comment]
> > > @pennywise94 I ran into this issue too; the giant solid happens because calling `showSkeleton()` on a `UITableView` calls `addDummyDataSource()`, which calls `reloadData()`, which immediately afterwards results in `visibleCells` returning an empty array. However, it turns out that [only visibleCells are skeleton'd](https://github.com/Juanpe/SkeletonView/blob/1671af6d2805211dc258f4b9b8574d228dd9efd7/Sources/SubviewsSkeletonables.swift#L18), so that explains why nothing is recursively skeleton'd below the `UITableView`. This definitely seems like a bug; if I get a chance I'll submit a PR to fix it.
> > > In the meantime, instead of calling `tableView.showSkeleton()` in `viewDidAppear()`, `tableView.visibleCells.forEach { $0.showSkeleton() }` worked for me. Let me know how that goes!
> > 
> > 
> > Hi mate, thanks for the reply. I eventually figured out that it had to do with the contentView for me. I added all my constraints to the contentView rather than the cell itself and it works like that. I also create a function addSkeleton() which is called in viewDidLayoutSubviews() and this works just fine for me.
> 
> thank u... finally it's work 😍😍😍

Glad I could help!

@erfanwakka Basically what I did was, instead of adding the views of my custom UITableViewCell directly like this:

`addSubview(testLabel)
testLabel.topAnchor.constraint(equalTo: topAnchor).isActive = true`

I would add it to the contentView, like this:

`contentView.addSubview(testLabel)
testLabel.topAnchor.constraint(equalTo: contentView.topAnchor).isActive = true`

And this resolved my problem.

**Comment by Juanpe** [comment]
Hi @pennywise94, I've added a new [illustration](https://github.com/Juanpe/SkeletonView#-collections) in the README file explaining how you should specify which elements are skeletonables using table views. Could you check if your config it's valid? I think this issue could be related to this.

**Comment by pennywise94** [comment]
Hi @Juanpe, first of all: many thanks for this great repository.

I have looked at the image and this is exactly how my project is set up. All my UITableViews are set up like you describe in the illustration. However, I didn't get SkeletonView to work with pinning my labels and views to the cell itself, such as:

`exampleLabel.topAnchor.constraint(equalTo: topAnchor, constant: 16).isActive = true`

For me, it started to work as soon as I replaced my anchor to the contentView, like this:

`exampleLabel.topAnchor.constraint(equalTo: contentView.topAnchor, constant: 16).isActive = true`

Of course I did this for all my anchors. I have then also set the contentView to be skeletonable, but I can now confirm that **removing `contentView.isSkeletonable = true` does not affect the SkeletonView**. So it is indeed not necessary to set the contentView to be skeletonable, but I find it **needs to be pinned to the contentView**. Is that correct?

Would enabling the skeleton for the contentView have significant downsizes for my (or any) project? I have now removed this, as it still works as expected.

Thank you for the feedback!
